hastexo/jobs.py

```diff
@@ -66,15 +66,32 @@ def run(self):
 
         # Get stacks to suspend
         with transaction.atomic():
-            stacks = Stack.objects.select_for_update().filter(
+            stacks_1 = Stack.objects.select_for_update().filter(
                 suspend_timestamp__isnull=False
+            ).filter(
+                suspend_by__isnull=True
             ).filter(
                 suspend_timestamp__lt=cutoff
             ).filter(
                 status__in=states
             ).exclude(
                 provider__exact=''
-            ).order_by('suspend_timestamp')[:concurrency]
+            )
+
+            stacks_2 = Stack.objects.select_for_update().filter(
+                suspend_timestamp__isnull=False
+            ).filter(
+                suspend_by__isnull=False
+            ).filter(
+                suspend_by__lt=timezone.now()
+            ).filter(
+                status__in=states
+            ).exclude(
+                provider__exact=''
+            )
+
+            stacks = stacks_1.union(stacks_2).order_by(
+                'suspend_timestamp')[:concurrency]
```

Comment:
I'm not quite sure what `[:concurrency]` does here, could you fill me in please?

Reply:
It limits the number of stacks to suspend in one job run.

Reply:
Ah! 💡 

Thanks!